### PR TITLE
Events support

### PR DESCRIPTION
**Pull request by anandijain**
I'll need to add the delays and the `useValuesFromTriggerTime` attribute later, but this is a good starting point for what i need done in the short term. Since it doesn't seem that the delays can be used easily in MTK yet, I can put that off for another PR.

### CODE REVIEW

**Review by exaexa — APPROVED** [review]
Looks good to me. We may wait for @giordano in case of any comments, but otherwise I think this is ready to merge.

File: src/structs.jl [hunk 1]
```diff
@@ -196,6 +196,29 @@ end
 """
 $(TYPEDEF)
 
+# Fields
+$(TYPEDFIELDS)
+"""
+struct Event
+    name
```
Comment (exaexa) [inline src/structs.jl:203]: naaaah, this needs types.

**Review by exaexa — CHANGES_REQUESTED** [review]

File: src/readsbml.jl [hunk 2]
```diff
@@ -467,6 +467,35 @@ function extract_model(mdl::VPtr)::SBML.Model
...
+    for n = 0:(num_events-1)
+        ev = ccall(sbml(:Model_getEvent), VPtr, (VPtr, Cuint), mdl, n)
+        sym = ccall(sbml(:Event_getId), Cstring, (VPtr,), ev)
+        trig = ccall(sbml(:Event_getTrigger), VPtr, (VPtr,), ev)
+        trig_math_ptr = ccall(sbml(:Trigger_getMath), VPtr, (VPtr,), trig)
+        if trig_math_ptr != C_NULL
+            trig_math = parse_math(trig_math_ptr)
+        end
+    
+        event_assignments = []
```
Comment (giordano) [inline src/readsbml.jl:482]:
```suggestion
        event_assignments = EventAssignment[]
```

**Review by giordano — COMMENTED** [review]
Looks good to me, apart from the missing type annotations

**Review by exaexa — APPROVED** [review]
cool thanks!

**Pull request merged**